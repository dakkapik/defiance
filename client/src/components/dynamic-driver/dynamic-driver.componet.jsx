import React from "react";
import "./dynamic-driver.styles.scss";
import MockImage from "./driver.jpg";
import DiscordCircle from "../discord-circle/discord-circle.component";
const DynamicDriver = ({
  userinfo: { firstName, employeeId, lastName, isActive },
}) => {
  return (
    <div className="sidecard">
      <div className="sidecard__avatar__container">
        <img alt="mock" className="sidecard__avatar" src={MockImage} />
        <DiscordCircle lightSwitch={!isActive} />
      </div>

      <div className="sidecard__content">
        <div className="sidecard__content__name">
          {firstName} {lastName}
        </div>
        <div className="sidecard__content__sub">
          <div className="sidecard__content__sub__text">
            <span>
              {employeeId}
              {/*props.position.timestamp*/}
            </span>
          </div>
        </div>
      </div>
    </div>
  );
};

export default DynamicDriver;
