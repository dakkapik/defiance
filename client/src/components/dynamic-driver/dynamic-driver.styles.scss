.sidecard {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  height: 52px;
  padding: 0 8px;
  margin-bottom: 10px;
  margin-left: 25px;

  &__avatar {
    border-radius: 100px;
    display: -webkit-box;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-box-flex: 0;
    -ms-flex: 0 0 auto;
    flex: 0 0 auto;
    width: 42px;
    height: 42px;
    margin-right: 12px;
  }

  &__content {
    white-space: nowrap;
    text-overflow: ellipsis;
    overflow: hidden;

    &__name {
      font-size: 16px;
      display: -webkit-box;
      display: -ms-flexbox;
      display: flex;
      -webkit-box-pack: start;
      -ms-flex-pack: start;
      justify-content: flex-start;
      -webkit-box-align: center;
      -ms-flex-align: center;
      align-items: center;
    }

    &__sub {
      margin-top: -2px;
      white-space: nowrap;
      text-overflow: ellipsis;
      overflow: hidden;
      &__text {
        display: -webkit-box;
        display: -ms-flexbox;
        display: flex;
        -webkit-box-align: center;
        -ms-flex-align: center;
        align-items: center;
        font-size: 12px;
        line-height: 16px;
        color: rgb(189, 187, 187);
        font-weight: 500;
      }
      &__text > span {
        margin-left: 8px;
      }
    }
  }
}
