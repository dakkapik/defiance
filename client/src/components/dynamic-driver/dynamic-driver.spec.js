import React from "react";
import { DynamicDriver } from "./dynamic-driver.componet";
import { shallow } from "enzyme";

describe("DynamicDriver component render", () => {
  let wrapper;

  beforeEach(() => {
    const mockProps = {
      userinfo: {
        firstName: "felipe",
        lastName: "rodas",
        employeeId: 3533,
        isActive: false,
      },
    };
    wrapper = shallow(<DynamicDriver {...mockProps} />);
  });

  it("should render DynamicDriver component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
