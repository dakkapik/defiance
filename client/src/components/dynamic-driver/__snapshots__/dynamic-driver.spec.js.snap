// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DynamicDriver component render should render DynamicDriver component 1`] = `ShallowWrapper {}`;
